src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -228,6 +242,31 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
             return base.UpdateValueCore(updateValueParams);
         }
 
+        /// <summary>
+        /// Updates the Script property of the node.
+        /// NOTE: This is a temporary method used during the Python 2 to Python 3 transistion period,
+        /// it will be removed when the transistion period is over.
+        /// </summary>
+        /// <param name="newCode">The new migrated code</param>
+        [Obsolete("Method will be deprecated after the Python 3 transition period")]
+        public void MigrateCode(string newCode)
+        {        
+            var e = new PythonCodeMigrationEventArgs(Script, newCode); 
+            Script = newCode;
+            OnCodeMigrated(e);
+        }
+
+        /// <summary>
+        /// Fires when the Script content is migrated to Python 3.
+        /// NOTE: This is a temporary event used during the Python 2 to Python 3 transistion period,
+        /// it will be removed when the transistion period is over.
+        /// </summary>
+        public event EventHandler CodeMigrated;
```

Comment:
Make type of `EventHandler<PythonCodeMigrationEventArgs>` to facilitate type safety on subscribers